packages/hyperion-autologging/src/ALSurfaceProxy.ts

```diff
@@ -56,8 +27,33 @@ type ProxyInitOptions =
     surfaceComponent: SurfaceComponent;
   }>;
 
+/**
+ * We need to use a hook to get the surface value, but the rules of using
+ * hooks won't allow us to call it outside of a react component.
+ * So, the following proxy component is purely for getting around this
+ * limitation and reading the current surface value during rendering.
+ * If we can find a way around this limitation, we can use a simpler logic
+ * like the following:
+ */
+function SurfaceProxy(props: React.PropsWithChildren<ProxyInitOptions>) {
```

Comment:
Was thinking too,  even though we have to wrap in another component it might be helpful as it will show these cases in the component stack more clearly instead of just another `Surface`.